salt/returners/couchdb_return.py

```diff
@@ -122,14 +193,19 @@ def _generate_doc(ret):
     return retc
 
 
-def _request(method, url, content_type=None, _data=None):
+def _request(method, url, content_type=None, _data=None, user=None, passwd=None):
     """
     Makes a HTTP request. Returns the JSON parse, or an obj with an error.
     """
     opener = _build_opener(_HTTPHandler)
     request = _Request(url, data=_data)
     if content_type:
         request.add_header("Content-Type", content_type)
+    if user and passwd:
+        auth_encode = "{0}:{1}".format(user, passwd).encode("base64")[:-1]
+        auth_basic = "Basic {0}".format(auth_encode)
```

Comment:
Pretty sure these format statements will need to have the numbers removed. So just `{}`